packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterAuthRule.kt

```diff
@@ -16,10 +16,74 @@
 package com.amazonaws.amplify.amplify_datastore.types.model
 
 import com.amplifyframework.core.model.AuthRule
+import com.amplifyframework.core.model.AuthStrategy
+import com.amplifyframework.core.model.ModelField
+import com.amplifyframework.core.model.ModelOperation
 
 data class FlutterAuthRule(val map: Map<String, Any>) {
+    
+    private val authStrategy : AuthStrategy =
+            indexToAuthStrategy(map["authStrategy"] as Int)
+    private val ownerField : String? = map["ownerField"] as String?
```

Comment:
In all of these cases, I don't think you need to say `String?` twice. Can't you just do:
```kotlin
private val ownerField = map["ownerField"] as String?
```

Also, if any of these are required, can you lose the `?`?

Reply:
Sounds good, I've updated all Kotlin files in types/model to not have these redundant type declarations

Yes that is true.  Those fields I found to be optional

Reply:
I feel in some cases it adds to the readability, e.g. when a variable is receiving value from a complex lambda expressions. To understand the type of the variable one would have to fully read and understand the complex transformations happening in the lambdas. I encountered that while reviewing this PR. That's just my perception though, happy to discuss.

Reply:
Sure let's discuss this.  The types in the Kotlin ModelSchema classes are not very complex and you can easily see their type by seeing our casting statement right after (as Int, as String, etc)